### PR TITLE
[libphonenumber] Ensure build reproducibility. JB#58635

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

File: rpm/libphonenumber.spec [hunk 1]
```diff
@@ -70,6 +72,7 @@ Contains files needed to development with %{name}.
        -DUSE_LITE_METADATA=ON \
        -DUSE_RE2=OFF \
        -DBUILD_STATIC_LIB=OFF \
+       -DCMAKE_BUILD_TYPE=Release \
```
Comment (martyone) [inline rpm/libphonenumber.spec]: This feels like a bug on its own that it needs to be enabled explicitly. Also worth to note it enables `-O3`.

Comment (martyone) [reply rpm/libphonenumber.spec]: Changed to `RelWithDebInfo`, which does not enables `-O3`. The leading note still applies - shouldn't that be the default?

Comment (vigejolla) [reply rpm/libphonenumber.spec]: Sure it feels like a bug, but then again CMake documentation (https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#default-and-custom-configurations) says "Users should always explicitly specify the build type" so this is in line with that.

**Review by vigejolla — APPROVED** [review]
LGTM

**Pull request merged**